src/main/scala/org/ergoplatform/nodeView/wallet/requests/TransactionSigningRequest.scala

```diff
@@ -0,0 +1,71 @@
+package org.ergoplatform.nodeView.wallet.requests
+
+import io.circe.{Decoder, Encoder, Json}
+import org.ergoplatform.http.api.ApiCodecs
+import org.ergoplatform.modifiers.mempool.UnsignedErgoTransaction
+import org.ergoplatform.wallet.secrets.{DhtSecretKey, DlogSecretKey, PrimitiveSecretKey}
+
+/**
+  * Basic trait for externally provided hint for interpreter (to be used during script execution and signature
+  * generation).
+  */
+trait Hint
+
+/**
+  * Externally provided secret (to be used once for a transaction to sign)
+  * @param key - the secret
+  */
+case class ExternalSecret(key: PrimitiveSecretKey) extends Hint
+
+/**
+  * A request to sign a transaction
+  * @param utx - unsigned transaction
+  * @param hints - hints for interpreter (such as additional one-time secrets)
+  * @param inputs - hex-encoded input boxes bytes for the unsigned transaction (optional)
+  * @param dataInputs - hex-encoded data-input boxes bytes for the unsigned transaction (optional)
+  */
+case class TransactionSigningRequest(utx: UnsignedErgoTransaction,
+                                     hints: Seq[Hint],
+                                     inputs: Option[Seq[String]],
```

Comment:
Can't the same semantics be encoded by just `Seq[String]`?

Reply:
@oskin1 @aslesarenko semantics of Option[Seq[...]] differs from Seq[...], and this is taken in account by the JSON (de)serializer. No need to comment here, denoting optional values via types is standard in Scala.

Reply:
@kushti what is the meaning of Some(Seq()), how it is different from None?

Reply:
@aslesarenko Some(Seq()) denotes existence of the empty sequence, None denotes absence of any collection in the value